econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -104,60 +105,67 @@ def _get_metadata_causal_insights_keys():
             _CausalInsightsConstants.ViewKey]
 
 
-def _first_stage_reg(X, y, *, automl=True):
+def _first_stage_reg(X, y, *, automl=True, random_state=None, verbose=0):
     if automl:
-        model = GridSearchCVList([make_pipeline(StandardScaler(), LassoCV()),
+        model = GridSearchCVList([make_pipeline(StandardScaler(), LassoCV(random_state=random_state)),
```

Comment:
Since now we are scaling the outcome, we should not use LassoCV, but rather just Lasso and add a grid of hyperparameters. This way we are selecting best on the CV of the grid search and not just running LassoCV for each fold. 

Then we also don't need to do the thing you have afterwards where we replace lassocv with lasso.

We could also use the sklearn function to generate the alpha grid:
https://github.com/scikit-learn/scikit-learn/blob/15a949460dbf19e5e196b8ef48f9712b72a3b3c3/sklearn/linear_model/_coordinate_descent.py#L81

Reply:
Scratch this. If we do sth like this, then the gcv, wont warm start lasso for each alpha point in the grid, and it will be much much slower than running lassocv. So let's leave the lassocv as is.

Reply:
ok